### PR TITLE
SOPHI-1367: Update iglu schema to accommodate fixes to snowplow-common-enrich

### PR DESCRIPTION
**Pull request by robertdecaire**
Because we changed the number of parameters in the setup config passed into the referer-parser module, the iglu schema can no longer validate the config. So we're directing it to a new schema, which is updated in https://github.com/globeandmail/sophi/pull/791.

@max-tgam pointed out that since we're already feeding the location of our iglu server to this module, all we should need to do is change the schema vendor and name, and it should pick up the new schema from our server.

### CODE REVIEW

**Review by dmatth — APPROVED** [review]
LGTM

**Comment by dmatth** [comment]
> it should pick up the new schema from our server

only if our server has it, which would only happen on a restart of Iglu, which we don't normally do. For now you would need to add the schema manually via our Iglu swagger API. The production version of that API is here: https://iglu.prod.sophi.io/ and Alireza can get you its API key.

**Comment by robertdecaire** [comment]
Yeah, Max mentioned that we'd need to do that manually. I will follow up after this has been merged.

**Review by max-tgam — APPROVED** [review]
LGTM.
Also you do not need to wait for this to merge to add schema to iglu manually.
Once https://github.com/globeandmail/sophi/pulls is merged - we can upload this. I can help you out on that - it takes 2 mins but there are a few things you have to be aware of when . you do it manually otherwise the enricher will not be able to retrieve the schema.

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/RefererParserEnrichment.scala
Comment (szareiangm) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/RefererParserEnrichment.scala]: can we increase the version of the snowplow one? i.e. `"com.snowplowanalytics.snowplow", "referer_parser", "jsonschema", 1, 1`?

**Review by szareiangm — CHANGES_REQUESTED** [review]
look good but if we are submitting this to snowplow, shall we rename the schema to the globe and mail?

**Comment by robertdecaire** [comment]
We're not submitting to Snowplow. I submitted two PRs to snowplow: 
https://github.com/snowplow-referer-parser/scala-referer-parser/pull/77/files
https://github.com/snowplow-referer-parser/referer-parser/pull/207/files

The first updates the Scala referer-parser directly to accept `android-app` as a URI scheme. If accepted, this change will be included in v0.6.0, which will be used by Snowplow R118.

The second updates the referers.yml file with some mobile app referers. If accepted, this will get in to v0.3.1, which will be used in Snowplow R117, but be rendered redundant in R118.

When R118 is available, with respect to referer parsing, I expect that we'll be able to revert these changes to our own repo and then upgrade to R118 (we're currently on R114), setting aside any modifications Max might have made. 

I think they're expecting that we might be able to use R117, and that there will be a long wait for R118, which is why they're pushing some unplanned interim changes like updating referers.yml. They've decided that adding in some fixes like that will be useful, and might help support us and other users who are waiting for the more substantial updates in R118.

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/RefererParserEnrichment.scala
Comment (robertdecaire) [reply 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/RefererParserEnrichment.scala]: No, because it's not actually the snowplow one. It's our own modified schema, which Snowplow is not going to use. If they ever updated it themselves, it would cause a version collision.

Comment (szareiangm) [reply 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/RefererParserEnrichment.scala]: then please remove `sophi_` from it.

Comment (robertdecaire) [reply 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/RefererParserEnrichment.scala]: Done.

**Comment by szareiangm** [comment]
here it becomes complicated. There is a ticket from Saud to override endpoints that we haven't submitted yet to Snowplow. (The other PR)

Can you cherry pick that commit and we build your branch for time being? any other solution is also appreciated.

**Comment by robertdecaire** [comment]
I've created a branch called `sophi-deployment` that's branched off of vanilla R114. We can reset `master` to `upstream/master`, and use `sophi-deployment` for "master" internal builds based on R114.

I've also pointed this PR at `sophi-deployment`, rather than `master`. It contains all the commits from SOPHI-1220, which had previously been merged into `master`.

**Comment by szareiangm** [comment]
I am confused here. 

There are three tickets done here: 

- SOPHI-1220
- SOPHI-1367
- And Saud's ticket

in our internal `sophi-deployment` branch, there should be three commits. 

and our master should be pointing to the master from snowplow upstream. because it originated from there and should be synced from there.

**Comment by szareiangm** [comment]
oh I see your other commits in this PR. Then `sophi-deployment` would have two commits. got it. thanks for helping.

**Comment by robertdecaire** [comment]
Yes, that's right. We'll lose track of 1220 in the commits unless we want to open a separate PR for those commits and squash them ahead of merging this one.

I don't think that's terribly important. Let's just squash this whole thing and end up with two commits in `sophi-deployment`.

**Comment by szareiangm** [comment]
works.

**Pull request merged**
